syntax = "proto2";

message n3rvmessage {
    required string sender = 1;
    required string action = 2;
    repeated string args = 3;
    required string payload = 4;
}

message n3rvdirectory {
    repeated n3rvnode nodes = 1;
}

message n3rvnode {
    required string namespace_ = 1;
    required string service_class = 2;
    required string name = 3;
    required string ip = 4;
    repeated n3rvbinding bindings = 6;
}

message n3rvbinding {
    required string name = 1;
    required int32 port = 2;
    required int32 socket_type = 3;
}